core/src/main/scala/org/apache/spark/util/AsynchronousListenerBus.scala

```diff
@@ -66,16 +66,13 @@ private[spark] abstract class AsynchronousListenerBus[L <: AnyRef, E](name: Stri
         self.synchronized {
           processingEvent = true
         }
+        if (stopped.get()) {
+          // Get out of the while loop and shutdown the daemon thread
+          return
+        }
```

Comment:
can you move this into the `try`? If `stopped.get` throws an exception we still want to set `processingEvent` to false